### PR TITLE
fix: Fetch app's current branch from redux store

### LINKED ISSUE #15692: [Bug]-[Infinity]:appsmith.store value not read on load (when branch query param is removed)
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When an app integrated to git is opened without the branch param.
Stored value using `storeValue("key", "value")` is not fetched on page load.

### Steps To Reproduce

1. Create an app integrated with git.
2. Create a button to store some value on click. -> `storeValue("value", "abc")`
3. Create a text box to show the stored value. -> `appsmith.store`
4. Deploy the app and re-open link **without branch query param**.
5. Click store value and refresh the page.
6. Notice `appsmith.store` is empty on load.

https://www.loom.com/share/88fc5c03acc6452893b5b846bb4bf209

### Public Sample App

_No response_

### Version

Cloud - 1.7.11

Front logo [Front conversations](https://app.frontapp.com/open/top_7gfbg)

### PR DESCRIPTION
**Pull request by ohansFavour**
### Cause of issue

Appsmith's persistent data is stored in the browser's localStorage under a key generated using the `applicationId` and `branch name`. When fetching the store's value on page load, if the branch's name is not present in the URL string, its value defaults to an emptyString.

### Solution
Default value of `branch name` should be the current git branch.

Fixes #15692 
## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
manual

### Test Plan
Cypress covers test case

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [x] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [x] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Comment by ramsaptami** [comment]
LGTM @ohansFavour

File: app/client/src/entities/Engine/index.ts [hunk 1]
```diff
@@ -69,7 +70,14 @@ export default abstract class AppEngine {
     if (!apiCalls)
       throw new PageNotFoundError(`Cannot find page with id: ${pageId}`);
     const application: ApplicationPayload = yield select(getCurrentApplication);
-    yield put(updateAppStore(getPersistentAppStore(application.id, branch)));
+    const currentGitBranch: string | undefined = yield select(
+      getCurrentGitBranch,
+    );
```
Comment (Rishabh-Rathod) [inline app/client/src/entities/Engine/index.ts:75]:
```suggestion
    const currentGitBranch: ReturnType<typeof getCurrentGitBranch> = yield select(
      getCurrentGitBranch,
    );
```

**Review by Rishabh-Rathod — APPROVED** [review]
Changes LGTM. 

Small suggestion.

File: app/client/src/entities/Engine/index.ts (hunk 1 above)
Comment (ohansFavour) [reply app/client/src/entities/Engine/index.ts]: Nice call out

**Review by Rishabh-Rathod — APPROVED** [review]
LGTM

**Pull request merged**